components/aws/sagemaker/tests/unit_tests/tests/TrainingJob/test_TrainingJob_spec.py

```diff
@@ -0,0 +1,34 @@
+from TrainingJob.src.TrainingJob_spec import SageMakerTrainingJobSpec
+import unittest
+
+
+class TrainingSpecTestCase(unittest.TestCase):
+
+    REQUIRED_ARGS = [
+        "--region",
+        "us-west-1",
+        "--algorithm_specification",
+        "{'trainingImage': '746614075791.dkr.ecr.us-west-1.amazonaws.com/sagemaker-xgboost:1.2-1', 'trainingInputMode': 'File'}",
+        "--hyper_parameters",
+        "{'max_depth': '2', 'gamma': '10', 'eta': '0.3', 'min_child_weight': '6', 'objective': 'multi:softmax', 'num_class': '10', 'num_round': '10'}",
+        "--input_data_config",
+        "[]",
+        "--output_data_config",
+        "{'s3OutputPath': 's3://ack-sagemaker-bucket-123456789012'}",
+        "--resource_config",
+        "{'instanceCount': 1, 'instanceType': 'ml.m4.xlarge', 'volumeSizeInGB': 5}",
+        "--role_arn",
+        "arn:aws:iam::123456789012:role/ack-sagemaker-execution-role",
+        "--stopping_condition",
+        "{'maxRuntimeInSeconds': 86400}",
+        "--training_job_name",
+        "kfp-ack-training-job-99999",
+    ]
+
+    def test_minimum_required_args(self):
```

Comment:
Would adding a test with incorrect inputs to verify the exception is raised be a large effort?

Reply:
done